packages/e2e-tests/specs/editor/blocks/navigation.test.js

```diff
@@ -211,6 +211,18 @@ async function getNavigationMenuRawContent() {
 // Disable reason - these tests are to be re-written.
 // eslint-disable-next-line jest/no-disabled-tests
 describe( 'Navigation', () => {
+	const contributorUsername = 'contributoruser';
+	let contributorPassword;
+
+	beforeAll( async () => {
+		// Creation of the contributor user **MUST** be at the top level describe block
+		// otherwise this test will become unstable. This action only happens once
+		// so there is no huge performance hit.
+		contributorPassword = await createUser( contributorUsername, {
+			role: 'contributor',
+		} );
```

Comment:
Maybe there's a bug in the e2e helpers so that switching to a newly created user is unreliable? Otherwise, I can't think of any other reason why this has to be at the root level.

`loginUser` doesn't check if the submission is successful, it only tries to fill out the login form and press the login button. Perhaps we can add an assertion in the helper and/or throw a meaningful error message as to why it fails.

The same goes `createUser`. It doesn't check if the action is successful either. The password it returns is auto-generated before the submission, so checking its existence isn't enough either.

I'm guessing maybe one of these helpers failed at some point and we didn't catch it. The reason it's less unreliable in top-level describe block is still unknown though.

Reply:
It'd definitely be good to try to understand the root cause, as otherwise it'll be difficult to use this utility in other tests.

Reply:
Probably a good opportunity to rewrite these utils with `rest` or something similar too 🙈 .

Reply:
Sounds good. Are we able to ship this fix now though so we have test coverage and then circle back to making it REST?

Reply:
Yes, absolutely. I have a draft PR opened at #38060 which might be helpful for that matter and I'll work on it next.